crates/oxc_cli/src/lib.rs

```diff
@@ -42,6 +42,10 @@ impl Cli {
                 Walk::new(path, &self.cli_options)
                     .iter()
                     .filter(|path| {
```

Comment:
I think all these ignore logic should all be moved to the walker, it's unrelated to this linter run anymore.

Reply:
https://docs.rs/ignore/0.4.20/ignore/struct.WalkBuilder.html#method.filter_entry
Does this work?

Reply:
The method needs the reference be `static`.
![image](https://user-images.githubusercontent.com/47748132/222342123-ca909e1e-8ab2-40ef-bf94-c8bd0b53b98f.png)

Reply:
I tried 
```rust
		let ignore_pattern = options.ignore_pattern.to_owned();
        if !options.no_ignore {
            inner.add_custom_ignore_filename(&options.ignore_path).filter_entry(move |dir_entry| {
                for pattern in ignore_pattern.iter() {
                    if pattern.matches_path(dir_entry.path()) {
                        return false;
                    }
                }
                true
            });
        }
```

But the output of `cargo run --bin oxc_cli lint --ignore-pattern "*.js"  test.js` still printed the lint result

Reply:
It seems that the `filter_entry` didn't call the closure. I write a `println!` in it, but it didn't print anything.

Reply:
I have no idea where the passed in `filter` will be called in [the source code](https://github.com/BurntSushi/ripgrep/blob/master/crates/ignore/src/walk.rs) 🥲

Reply:
Let's figure it out later then 😄

Reply:
I found that the `filter` will be called in the [`skip_entry`](https://github.com/BurntSushi/ripgrep/blob/master/crates/ignore/src/walk.rs#L962), and this method is called in [here](https://github.com/BurntSushi/ripgrep/blob/master/crates/ignore/src/walk.rs#L1009-L1029C55). The `filter` closure should be called when we used the `Walk`, but it didn't. 🤔

Reply:
Let's give up for now 😆 This `ignore` interface seems way too complicated

Reply:
Okay! If we don't use the `skip_entry` but still want to handle all ignore logic in `walk.rs`, we can store the ignore related variable in `Walk` struct. Then deal with them in the `iter` method. Would you like me do that? Or you have another idea.

Reply:
Let's differ it, I want to look at why `skip_entry` doesn't work